### PR TITLE
install java in pre-baked codeceptjs image

### PR DESCRIPTION
**Pull request by PeterNgTr**
## Motivation/Description of the PR

Applicable helpers:

- [ ] Webdriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe

- May fix this https://github.com/Codeception/CodeceptJS/issues/1928

## Type of change

- [ ] Breaking changes
- [ ] New functionality
- [x] Bug fix
- [ ] Documentation changes/updates
- [ ] Hot fix
- [ ] Markdown files fix - not related to source code

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `robo docs`)
- [ ] Lint checking (Run `npm run lint`)
- [ ] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -28,6 +28,22 @@ RUN apt-get update && apt-get install -y wget --no-install-recommends \
...
+# Fix certificate issues
+RUN apt-get update && \
+    apt-get install ca-certificates-java && \
+    apt-get clean && \
+    update-ca-certificates -f;
+
+# Setup JAVA_HOME -- useful for docker commandline
+ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64/
+RUN export JAVA_HOME
+RUN java -version
```
Comment (Vorobeyko) [inline Dockerfile]: This script to add additional  layout and may increases size)
Add this line to previous line, please)

File: Dockerfile [hunk 2]
```diff
@@ -28,6 +28,22 @@ RUN apt-get update && apt-get install -y wget --no-install-recommends \
...
 # ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD true
 
+# Install OpenJDK-8
+RUN apt-get update && \
+    apt-get install -y openjdk-8-jdk && \
+    apt-get install -y ant && \
+    apt-get clean;
+
+# Fix certificate issues
+RUN apt-get update && \
```
Comment (Vorobeyko) [inline Dockerfile]: And may be this move to previous script?

File: Dockerfile (hunk 1 above)
Comment (PeterNgTr) [reply Dockerfile]: well, I forgot to remove that line. Just want to check if the java is installed 😄

**Comment by Vorobeyko** [comment]
What problem fix this MR?

**Comment by PeterNgTr** [comment]
> What problem fixes with this "MR"?

this issue https://github.com/Codeception/CodeceptJS/issues/1928

**Comment by DavertMik** [comment]
Sorry, I didn't get it - where Java is used here?

**Comment by PeterNgTr** [comment]
This is due to the docker image not having a java runtime installed. I am able to test and fix the issue by installing a jre in the container on startup.

Obviously this can be fixed by creating your own docker image using the one you guys provide as a base but it might make sense to include a JRE in the base image as well, or at least provide a separate image on a specific tag that does contain it.

Having to manage your own Docker image and keep it up to date when you release new versions can be a tedious task so it would be a lot more convenient if there was a pre-baked image that had everything needed to run Webdriver based tests.

**Comment by Vorobeyko** [comment]
Why webdriver use java? 
Maybe need in configuration file of webdriverio add chrome executablePath?

**Comment by PeterNgTr** [comment]
> Why webdriver use java?
> Maybe need in configuration file of webdriverio add chrome executablePath?
> 
> or did in image use selenium?

well, I guess there is a case when `selenium-standalone` wdio plugin is used, it needs java but the our docker image is built from `node` which has no pre-installed java.

**Comment by DavertMik** [comment]
Sorry, for Java and selenium you should use different docker container and connect them via docker compose. I don't think Java should be here

**Comment by PeterNgTr** [comment]
Okay. Then I‘ll close it, thou.

**Comment by AllysonYuShiong** [comment]
@PeterNgTr I need a fix for this issue. Is there any workaround?

**Comment by PeterNgTr** [comment]
Hey @AllysonYuShiong you could try building your image based on codeceptjs and adding java installation.

**Comment by AllysonYuShiong** [comment]
@PeterNgTr I tried to build with the command `docker build -t codeceptjs/codeceptjs`. With Dockerfile having the install java command in the commit, I have received the error
Screenshot 2021-06-29 at 4 04 10 PM

**Comment by AllysonYuShiong** [comment]
@PeterNgTr is adding java installation on the Dockerfile in codeceptjs? can you provide me instructions on how to have java pre-installed?

**Comment by AllysonYuShiong** [comment]
I tried to use selenium/standalone-chrome-debug as per discussed in https://codecept.discourse.group/t/docker-with-webdriver-io/447 and I still can get the error `not found: java` when I run the test.

**Pull request closed without merging**